src/cmd/build.js

```diff
@@ -40,11 +42,43 @@ export type PackageCreatorParams = {
 export type PackageCreatorFn =
     (params: PackageCreatorParams) => Promise<ExtensionBuildResult>;
 
+export type LocalizedNameParams = {
+  manifestData: ExtensionManifest,
+  messageData: any,
+};
+
+export function getDefaultLocalizedName(
+  {manifestData, messageData}: LocalizedNameParams
+  ): string {
+  let extensionName: string;
+
+  if (messageData) {
+    try {
+      messageData = JSON.parse(messageData);
+    }
+    catch (err) {
+      err.message = 'The JSON file is malformed';
+      let error = new WebExtError(err.message);
```

Comment:
In this situation you want to preserve the original error but also make the error more readable. I would suggest an approach like this:

    throw new WebExtError(`The JSON file is malformed: ${err}`);

This will throw something like "The JSON file is malformed: SyntaxError: unexpected token at column 45"